packages/scene-composer/src/components/three-fiber/MotionIndicatorComponent/MotionIndicatorComponent.tsx

```diff
@@ -72,8 +72,10 @@ const MotionIndicatorComponent: React.FC<IMotionIndicatorComponentProps> = ({
         dataBindingTemplate,
       );
 
-    const result = ruleEvaluator(0, values, speedRule) as number;
-    return result >= 0 ? result : 0;
+    const result = ruleEvaluator(0, values, speedRule);
```

Comment:
shouldn't it be `ruleEvaluator(0, values, speedRule).target`?

Reply:
We are returning returning the RuleReturnResult and there is only single value in it. This is also one way of getting the returned result?

Reply:
but `ruleEvaluator` returns `{ target: xx, targetMetadata: yy}`, why would `typeof result === 'number'` be true?

Reply:
I removed targetMetadata from this. That is not required here. If (evalResult) type is number  we are returning Number(evalResult) so here in this rule evaluator speed is a number.